util/top_gen/generator.py

```diff
@@ -364,8 +344,8 @@ def legacy_ordering(input_pin: OutputBlockIo) -> tuple[int, int, int]:
         ("rtl/templates/pinmux.sv.tpl", "rtl/system/pinmux.sv"),
         ("doc/ip/pinmux.md.tpl", "doc/ip/pinmux.md"),
     ):
-        print("Generating from template: " + template_file)
+        print("Generating from template:", template_file)
         content = Template(filename=template_file).render(**template_variables)
-        Path(output_file).write_text(content)
+        Path(output_file).write_text(str(content))
```

Comment:
Why was this necessary?

Reply:
The `pyright` type checker will complains that `Template` could return types other than string. This makes it happy.

Reply:
Cool well thanks for fixing this.